snuba/query/allocation_policies/cross_org.py

```diff
@@ -0,0 +1,172 @@
+            required_tenant_types:
+              - referrer
+            default_config_overrides:
+              is_enforced: 0
+              is_active: 0
+            cross_org_referrer_limits:
+              dynamic_sampling.counters.get_org_transaction_volumes:
+                max_threads: 4
+                concurrent_limit: 10
+    ```
+
+    Each referrer gets a concurrent limit (applied per referrer) and a max_threads limit (applied to every query made by that referrer).
+    Both limits are static but changeable at runtime
+    """
+
+    @property
+    def rate_limit_name(self) -> str:
+        return "cross_org_query_policy"
+
+    def _additional_config_definitions(self) -> list[AllocationPolicyConfig]:
+        return super()._additional_config_definitions() + [
+            AllocationPolicyConfig(
+                name="referrer_concurrent_override",
+                description="""override the concurrent limit for a referrer""",
+                value_type=int,
+                param_types={"referrer": str},
+                default=-1,
+            ),
+            AllocationPolicyConfig(
+                name="referrer_max_threads_override",
+                description="""override the max_threads for a referrer, applies to every query made by that referrer""",
+                param_types={"referrer": str},
+                value_type=int,
+                default=-1,
+            ),
+        ]
+
+    def _validate_cross_org_referrer_limits(
+        self, cross_org_referrer_limits: dict[str, dict[str, int]]
+    ) -> None:
+        for referrer, limits in cross_org_referrer_limits.items():
+            concurrent_limit = limits.get("concurrent_limit", None)
+            max_threads = limits.get("max_threads", None)
+            if not isinstance(max_threads, int):
+                raise ValueError(
+                    f"max_threads is required for {referrer} and must be an int"
+                )
+            if not isinstance(concurrent_limit, int):
+                raise ValueError(
+                    f"concurrent_limit is required for {referrer} and must be an int"
+                )
+
+    def __init__(
+        self,
+        storage_key: StorageKey,
+        required_tenant_types: list[str],
+        default_config_overrides: dict[str, Any],
+        **kwargs: str,
+    ) -> None:
+        super().__init__(
+            storage_key, required_tenant_types, default_config_overrides, **kwargs
+        )
+        self._registered_cross_org_referrers = cast(
+            "dict[str, dict[str, int]]", kwargs.get("cross_org_referrer_limits", {})
+        )
+        self._validate_cross_org_referrer_limits(self._registered_cross_org_referrers)
+
+    def _get_quota_allowance(
+        self, tenant_ids: dict[str, str | int], query_id: str
+    ) -> QuotaAllowance:
+        referrer = str(tenant_ids.get("referrer", "no_referrer"))
+        referrer_is_registered = referrer in self._registered_cross_org_referrers
```

Comment:
I mentioned this in a comment above, but couldn't the referrer be automatically included if `cross_org_query=1`? Why do we need that flag and this check?